app/components/blacklight/search_context/client_applied_params_component.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+module Blacklight
+  module SearchContext
+    class ClientAppliedParamsComponent < Blacklight::Component
+      delegate :search_action_path, to: :helpers
+      def render?
+        controller.params.permit(:counter)[:counter]
```

Comment:
Is there a reason you need `.permit(:counter)` here?

Reply:
I misunderstood the source of the unpermitted parameter warnings I was getting, but I see they are from search state on show actions. I'll remove the two places this pattern is used in this PR.